relay-server/src/utils/error_boundary.rs

```diff
@@ -1,45 +1,12 @@
-use std::fmt::{self, Debug};
+use std::error::Error;
+use std::sync::Arc;
 
-use failure::Fail;
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 use serde_json::Value;
 
-/// Structure used in the implementation of Clone for ErrorBoundary
-/// Couldn't find a nicer way to do it.
-/// For the case where ErrorBoundary contains an error (Err enum) I copy
-/// the representation of the Fail trace object inside a ClonedFail object
-/// during a clone, that means that I loose the original Fail object and only
-/// retain the text message.
-#[derive(Fail, Debug, Clone)]
-struct ClonedFail(String);
-
-impl fmt::Display for ClonedFail {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}\n(cloned)", self.0)
-    }
-}
-
-/// Helper trait used to convert any Fail dyn object into a ClonedFail struct
-trait AsClonedFail {
-    fn as_cloned_fail(&self) -> ClonedFail;
-}
-
-impl AsClonedFail for ClonedFail {
-    /// if the object is already a ClonedFail just clone the message
-    fn as_cloned_fail(&self) -> ClonedFail {
-        self.clone()
-    }
-}
-
-impl AsClonedFail for dyn Fail {
-    fn as_cloned_fail(&self) -> ClonedFail {
-        ClonedFail(self.to_string())
-    }
-}
-
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub enum ErrorBoundary<T> {
-    Err(Box<dyn Fail>),
+    Err(Arc<dyn Error + Send + Sync + 'static>),
```

Comment:
Turns out we can clone the error if we put the error behind an `Arc`. Since the error boundary is used in some types that are sent between services or used in async functions, I additionally constrained to `Send + Sync`.